plugins/org.eclipse.glsp.graph/src/org/eclipse/glsp/graph/impl/GModelIndexImpl.java

```diff
@@ -118,7 +119,11 @@ protected void handleContainment(final Notification notification) {
 
    @Override
    public Optional<GModelElement> get(final String elementId) {
-      return Optional.ofNullable(idToElement.get(elementId));
+      Optional<GModelElement> indexMatch = Optional.ofNullable(idToElement.get(elementId));
+      if (indexMatch.isPresent() || !isCurrentlyBuildingIndex()) {
```

Comment:
My brain has a hard time parsing OR... NOT, I find the other way usually easier, e.g., isEmpty AND isCurrentlyBuilding..., but that is probably just me ;-)

Reply:
Yeah, I agree... I've now changed it to `!isPresent && isCurrentlyBuildingIndex()`. We can't use isEmpty because of the Java-8 compatibility. But still, I think it is better this way. Thanks!